diff -r -U 5 orig_srss_files/opt/SUNWut/lib/dhcp/isc/utdhcpservice /opt/SUNWut/lib/dhcp/isc/utdhcpservice
--- orig_srss_files/opt/SUNWut/lib/dhcp/isc/utdhcpservice	2007-08-02 00:44:01.000000000 +0200
+++ /opt/SUNWut/lib/dhcp/isc/utdhcpservice	2008-05-10 15:23:03.000000000 +0200
@@ -89,17 +89,23 @@
 exit 1
 }
 
 function StatusDHCP {
     typeset PACKAGE="${DHCP_PACKAGE}"
-    if $DHCP_INSTALLED ; then
-      # get the actual package name including version and release
-      PACKAGE="$(rpm -q ${DHCP_PACKAGE} 2> /dev/null)"
-    fi
-    if ! $DHCP_RUNNING && [ $DHCP_STATE != "unconfigured" ] ; then
-      DHCP_STATE="disabled"
-    fi
+    DHCP_INSTALLED=true
+    DHCP_ENABLED=true
+    DHCP_CONFIGURED=true
+    DHCP_RUNNING=true
+    DHCP_PACKAGE="dhcp3-server"
+    DHCP_STATE="online"
+    #if $DHCP_INSTALLED ; then
+    #  # get the actual package name including version and release
+    #  PACKAGE="$(rpm -q ${DHCP_PACKAGE} 2> /dev/null)"
+    #fi
+    #if ! $DHCP_RUNNING && [ $DHCP_STATE != "unconfigured" ] ; then
+    #  DHCP_STATE="disabled"
+    #fi
     print "begin dhcpstate"
     print "installed=${DHCP_INSTALLED}"
     print "enabled=${DHCP_ENABLED}"
     print "configured=${DHCP_CONFIGURED}"
     print "running=${DHCP_RUNNING}"
diff -r -U 5 orig_srss_files/opt/SUNWut/lib/utadmingid /opt/SUNWut/lib/utadmingid
--- orig_srss_files/opt/SUNWut/lib/utadmingid	2007-08-02 00:44:00.000000000 +0200
+++ /opt/SUNWut/lib/utadmingid	2008-05-10 15:23:03.000000000 +0200
@@ -13,9 +13,9 @@
 # back to using group 'root' as the directory owner.
 #
 ETCDIR="/etc/opt/SUNWut"
 UTADMINPW=${ETCDIR}/utadmin.pw
 if [ -f $UTADMINPW ] ; then
-	WEBGUI_GROUP=`/bin/ls -gn $UTADMINPW | /bin/awk '{print $3}' `
+	WEBGUI_GROUP=`/bin/ls -gn $UTADMINPW | /usr/bin/awk '{print $3}' `
 fi
 WEBGUI_GROUP=${WEBGUI_GROUP:-root}
 print $WEBGUI_GROUP
diff -r -U 5 orig_srss_files/opt/SUNWut/lib/utdmsession /opt/SUNWut/lib/utdmsession
--- orig_srss_files/opt/SUNWut/lib/utdmsession	2007-08-02 00:43:54.000000000 +0200
+++ /opt/SUNWut/lib/utdmsession	2008-05-10 16:52:35.000000000 +0200
@@ -21,14 +21,14 @@
 #	utdmsession -d Xdisplay [-z tag]
 #	    destroy DM session based on X display
 #	    -z - optional tag for logging purposes
 #
 
-#exec 2>&1 2>/var/tmp/utdmsession.$$       # Debug
-#set -x
+exec 2>&1 2>/var/tmp/utdmsession.$$       # Debug
+set -x
 
-MOD="`/bin/basename $0`"
+MOD="`basename $0`"
 USAGE="usage: $MOD [-c|-d] Xdisplay [-z tag]"
 UTMNT_DIRLOCK=".session"
 
     ID="/usr/bin/id"
     DPY=""
diff -r -U 5 orig_srss_files/opt/SUNWut/lib/utprodinfo /opt/SUNWut/lib/utprodinfo
--- orig_srss_files/opt/SUNWut/lib/utprodinfo	2007-08-02 01:45:16.000000000 +0200
+++ /opt/SUNWut/lib/utprodinfo	2008-05-10 15:23:03.000000000 +0200
@@ -337,11 +337,12 @@
 	buildParamList "$@"
 	if [[ $? -eq 1 ]]; then
 		# found no params, just return
 		return 0
 	fi
-	rpm -q --queryformat ${MAPPEDLIST} $PKGNAME 2>/dev/null
+	echo "/opt"
+	#rpm -q --queryformat ${MAPPEDLIST} $PKGNAME 2>/dev/null
 	return 0
 }
 
 
 #
@@ -352,10 +353,11 @@
 Linux_testPkg() {
 	$DEBUG
 	if [[ $# -ne 2 ]]; then
 		return 1
 	fi
+	return 0
 	case $1 in
 	"installed")	# package installed, could be either partial or complete
 		rpm -q --quiet $2 > /dev/null 2>&1
 		return $?;;
 	"partial")	# package partially installed
diff -r -U 5 orig_srss_files/opt/SUNWut/lib/xmgr/gdm/notify /opt/SUNWut/lib/xmgr/gdm/notify
--- orig_srss_files/opt/SUNWut/lib/xmgr/gdm/notify	2007-08-02 00:43:58.000000000 +0200
+++ /opt/SUNWut/lib/xmgr/gdm/notify	2008-05-10 17:38:16.000000000 +0200
@@ -7,11 +7,11 @@
 # Script to notify gdm that the Xservers file and  Xconfig file have changed
 #
 
 verbose=0
 umask 0022
-DMNAME=gdm-binary
+DMNAME=gdm
 unset LD_LIBRARY_PATH
 BASE=/etc/opt/SUNWut/basedir
 SUNWUTLIB=$BASE/lib
 XSUN=$SUNWUTLIB/utxsun
 
diff -r -U 5 orig_srss_files/opt/SUNWut/lib/xmgr/gdm/reset-dpy /opt/SUNWut/lib/xmgr/gdm/reset-dpy
--- orig_srss_files/opt/SUNWut/lib/xmgr/gdm/reset-dpy	2007-08-02 00:43:58.000000000 +0200
+++ /opt/SUNWut/lib/xmgr/gdm/reset-dpy	2008-05-10 17:38:47.000000000 +0200
@@ -8,11 +8,11 @@
 # Pass SRSS config directory as first arg (e.g. reset-dpy /tmp/SUNWut/config)
 #
 
 CONFIGDIR=$1
 CONFIGXDIR=$CONFIGDIR/xconfig
-DMNAME=gdm-binary
+DMNAME=gdm
 
 # Note - we start MINDISP at 11 using gdm for linux. It gets around bug
 # 5057552, which is actually caused by lax privileges defined by 
 # /etc/security/console.perms
 MINDISP=11              # minimum display number XXX
@@ -42,11 +42,12 @@
 #
 
 pgrep -P 1 -u 0 $DMNAME >/dev/null
 if [[ $? -eq 0 ]]
 then
-	dpys=`/usr/bin/gdmdynamic -l | /bin/sed -e 's/:\([^,]*\),[^;]*/\1/g' -e 's/;/ /g' `
+    dpys=`/usr/bin/gdmdynamic -l | tr ";" "\012" | sed -e "s/:\(.*\),.*,.*/\1/" | tr "\012" " "`
+	#dpys=`/usr/bin/gdmdynamic -l | /bin/sed -e 's/:\([^,]*\),[^;]*/\1/g' -e 's/;/ /g' `
 	for dpy in $dpys
 	do
           if [[ $dpy -ge $MINDISP && $dpy -le $MAXDISP ]]; then
             /usr/bin/gdmdynamic -d $dpy >/dev/null
           fi
diff -r -U 5 orig_srss_files/opt/SUNWut/sbin/utadm /opt/SUNWut/sbin/utadm
--- orig_srss_files/opt/SUNWut/sbin/utadm	2007-08-02 00:44:01.000000000 +0200
+++ /opt/SUNWut/sbin/utadm	2008-05-10 15:31:46.000000000 +0200
@@ -217,13 +217,13 @@
 		if [[ -r "/etc/networks" ]] ; then
 		  NETWORKS="/etc/networks";
 		else
 		  NETWORKS="${ETC_OPT_UT}/net/networks";
 		fi
-		DHCPCONFIG="/etc/dhcpd.conf"
+		DHCPCONFIG="/etc/dhcp3/dhcpd.conf"
 		GREP=grep	# XXX need to figure out which <===
-		INIT_DHCP="/etc/init.d/dhcpd";
+		INIT_DHCP="/etc/init.d/dhcp3-server";
 		if [[ -d "/etc/sysconfig/network-scripts" ]] ; then
 		  IFCONFIG_SCRIPT="/etc/sysconfig/network-scripts/ifcfg-"
 		  IFCONFIG_BOOT="ONBOOT=yes"
 		  BOOTPROTO=none
 		else
@@ -3149,15 +3149,16 @@
 }
 
 CountIPLeases() {
 	tmp_leases=/var/opt/SUNWut/tmp/tmpLeases.$$
 	dhcpd_hosts=/var/opt/SUNWut/tmp/dhcpdHosts.$$
-	if [ -f /var/lib/dhcp/dhcpd.leases ]; then
-		lease_file=/var/lib/dhcp/dhcpd.leases
-	elif [ -f /var/lib/dhcp/db/dhcpd.leases ]; then
-                lease_file=/var/lib/dhcp/db/dhcpd.leases
-        fi
+	lease_file=/var/lib/dhcp3/dhcpd.leases
+#	if [ -f /var/lib/dhcp/dhcpd.leases ]; then
+#		lease_file=/var/lib/dhcp/dhcpd.leases
+#	elif [ -f /var/lib/dhcp/db/dhcpd.leases ]; then
+#                lease_file=/var/lib/dhcp/db/dhcpd.leases
+#        fi
 
 	rm -f $tmp_leases
         rm -f $dhcpd_hosts
 
 	process_leases
diff -r -U 5 orig_srss_files/opt/SUNWut/sbin/utconfig /opt/SUNWut/sbin/utconfig
--- orig_srss_files/opt/SUNWut/sbin/utconfig	2007-08-02 00:44:01.000000000 +0200
+++ /opt/SUNWut/sbin/utconfig	2008-05-10 15:23:03.000000000 +0200
@@ -1558,13 +1558,13 @@
 . ${G_MEDIA_DIR}/support_lib/sras_config
 fi
 UT_PROD_NAME="Sun Ray Core Services"
 UT_VERSION="4.0"
 
-if ! UTInstalled; then
-  Fatal "$UT_PROD_NAME is not installed on this host"
-fi
+#if ! UTInstalled; then
+#  Fatal "$UT_PROD_NAME is not installed on this host"
+#fi
 
 UTA_BASEDIR="$(${G_MEDIA_DIR}/utprodinfo -r SUNWuta)/SUNWut"
 UTO_BASEDIR="$(${G_MEDIA_DIR}/utprodinfo -r SUNWuto)/SUNWut"
 
 # autoconfigure doesn't affect kiosk
@@ -1590,27 +1590,27 @@
 
 SRDS_PN="Sun Ray Data Store"
 SRDS_RN="3.0"
 SRDS_PACKAGE="SUNWutdso"
 
-if ! SRDSInstalled; then
-  Fatal "$SRDS_PN is not installed on this host"
-elif ! SRDSVersion $SRDS_RN; then
-  Fatal "$SRDS_PN $SRDS_RN is not installed on this host"
-elif SRDSInstallPartial; then
-  Fatal "$SRDS_PN $SRDS_RN is only partially installed"
-fi
+#if ! SRDSInstalled; then
+#  Fatal "$SRDS_PN is not installed on this host"
+#elif ! SRDSVersion $SRDS_RN; then
+#  Fatal "$SRDS_PN $SRDS_RN is not installed on this host"
+#elif SRDSInstallPartial; then
+#  Fatal "$SRDS_PN $SRDS_RN is only partially installed"
+#fi
 
 SRDS_BASEDIR="$(${G_MEDIA_DIR}/utprodinfo -r $SRDS_PACKAGE)/SUNWut"
 
 LCL_PROD_NAME="LDAP Client Libraries"
 
-if ! LdapClientInstalled; then
-  Fatal "$LCL_PROD_NAME is not installed on this host"
-elif LdapClientInstallPartial; then
-  Fatal "$LCL_PROD_NAME is only partially installed"
-fi
+#if ! LdapClientInstalled; then
+#  Fatal "$LCL_PROD_NAME is not installed on this host"
+#elif LdapClientInstallPartial; then
+#  Fatal "$LCL_PROD_NAME is only partially installed"
+#fi
 
 UT_ROOT_S="Sun Ray root"
 DEFAULT_ROOTENTRY="o=utdata"
 
 ROOTENTRY=""
diff -r -U 5 orig_srss_files/opt/SUNWut/sbin/utreplica /opt/SUNWut/sbin/utreplica
--- orig_srss_files/opt/SUNWut/sbin/utreplica	2007-08-02 00:44:01.000000000 +0200
+++ /opt/SUNWut/sbin/utreplica	2008-05-10 15:23:03.000000000 +0200
@@ -1750,41 +1750,41 @@
 fi
 
 # Check for SRS
 UT_PROD_NAME="Sun Ray Core Services"
 UT_VERSION="4.0"
-if ! UTInstalled; then
-   Fatal "$UT_PROD_NAME is not installed on this host"
-fi
+#if ! UTInstalled; then
+#   Fatal "$UT_PROD_NAME is not installed on this host"
+#fi
 
 # Check for SDS
 UTA_BASEDIR="$($UTPRODDIR/utprodinfo -r SUNWuta)/SUNWut"
 UTO_BASEDIR="$($UTPRODDIR/utprodinfo -r SUNWuto)/SUNWut"
 ETC_OPT_UT="/etc/opt/SUNWut"
 UT_TEMPLATE_LDAP="$UTA_BASEDIR/etc/template/ldap"
 SRDS_NAME="Sun Ray Data Store"
 SRDS_REL="3.0"
 SRDS_PACKAGE="SUNWutdso"
-if ! SunDSInstalled; then
-  Fatal "$SRDS_NAME is not installed on this host"
-elif ! SunDSVersion $SRDS_REL; then
-  Fatal "$SRDS_NAME $SRDS_REL is not installed on this host"
-elif SunDSInstallPartial; then
-  Fatal "$SRDS_NAME $SRDS_REL is only partially installed"
-fi
+#if ! SunDSInstalled; then
+#  Fatal "$SRDS_NAME is not installed on this host"
+#elif ! SunDSVersion $SRDS_REL; then
+#  Fatal "$SRDS_NAME $SRDS_REL is not installed on this host"
+#elif SunDSInstallPartial; then
+#  Fatal "$SRDS_NAME $SRDS_REL is only partially installed"
+#fi
 
 
 # check for LDAP Client libs
 LCL_PROD_NAME="LDAP Client Libraries"
 
-if [ -z $LCL_PACKAGE ]; then
-   Fatal "$LCL_PROD_NAME is not installed on this host"
-elif ! LdapClientInstalled; then
-   Fatal "$LCL_PROD_NAME is not installed on this host"
-elif LdapClientInstallPartial; then
-   Fatal "$LCL_PROD_NAME is only partially installed"
-fi
+#if [ -z $LCL_PACKAGE ]; then
+#   Fatal "$LCL_PROD_NAME is not installed on this host"
+#elif ! LdapClientInstalled; then
+#   Fatal "$LCL_PROD_NAME is not installed on this host"
+#elif LdapClientInstallPartial; then
+#   Fatal "$LCL_PROD_NAME is only partially installed"
+#fi
 
 # Ensure that utconfig has been utilised
 if [[ ! -f $UT_ADM_CONF ]]; then
   Fatal "please run utconfig prior to using utreplica"
 fi
