diff -ru tmp.orig/utadem/Makefile tmp/utadem/Makefile
--- tmp.orig/utadem/Makefile	2009-10-16 13:53:27.000000000 +0200
+++ tmp/utadem/Makefile	2009-10-16 13:41:13.000000000 +0200
@@ -5,7 +5,9 @@
 # Use is subject to license terms.
 #
 
-VERSION = $(shell uname -r)
+ifeq ($(VERSION),)
+	VERSION = $(shell uname -r)
+endif
 ISABITS_CMD = case "`/bin/uname -m`" in x86_64) echo 64;; i?86*|i86*) echo 32;; esac
 ISAINFO = $(shell $(ISABITS_CMD))
 ISA32_SPECIFIC_CPPFLAGS=-m32 -D__WORDSIZE=32
@@ -40,7 +42,7 @@
 
 # make invoked directly
 
-KDIR	:= /lib/modules/$(shell uname -r)/build
+KDIR	:= /lib/modules/$(VERSION)/build
 PWD	:= $(shell pwd)
 
 KOBJ = $(C_FILES:%.c=%.ko)
Only in tmp/utadem: Module.symvers
diff -ru tmp.orig/utdisk/Makefile tmp/utdisk/Makefile
--- tmp.orig/utdisk/Makefile	2009-10-16 13:53:27.000000000 +0200
+++ tmp/utdisk/Makefile	2009-10-16 13:40:10.000000000 +0200
@@ -5,7 +5,9 @@
 # Use is subject to license terms.
 #
 
-VERSION = $(shell uname -r)
+ifeq ($(VERSION),)
+	VERSION = $(shell uname -r)
+endif
 ISABITS_CMD = case "`/bin/uname -m`" in x86_64) echo 64;; i?86*|i86*) echo 32;; esac
 ISAINFO = $(shell $(ISABITS_CMD))
 ISA32_SPECIFIC_CPPFLAGS=-m32 -D__WORDSIZE=32
@@ -39,7 +41,7 @@
 
 # make invoked directly
 
-KDIR	:= /lib/modules/$(shell uname -r)/build
+KDIR	:= /lib/modules/$(VERSION)/build
 PWD	:= $(shell pwd)
 
 KOBJ = $(C_FILES:%.c=%.ko)
Only in tmp/utdisk: Module.symvers
diff -ru tmp.orig/utio/Makefile tmp/utio/Makefile
--- tmp.orig/utio/Makefile	2009-10-16 13:53:27.000000000 +0200
+++ tmp/utio/Makefile	2009-10-16 13:41:54.000000000 +0200
@@ -5,7 +5,9 @@
 # Use is subject to license terms.
 #
 
-VERSION = $(shell uname -r)
+ifeq ($(VERSION),)
+	VERSION = $(shell uname -r)
+endif
 ISABITS_CMD = case "`/bin/uname -m`" in x86_64) echo 64;; i?86*|i86*) echo 32;; esac
 ISAINFO = $(shell $(ISABITS_CMD))
 ISA32_SPECIFIC_CPPFLAGS=-m32 -D__WORDSIZE=32
@@ -40,7 +42,7 @@
 
 # make invoked directly
 
-KDIR	:= /lib/modules/$(shell uname -r)/build
+KDIR	:= /lib/modules/$(VERSION)/build
 PWD	:= $(shell pwd)
 
 KOBJ = $(C_FILES:%.c=%.ko)
Only in tmp/utio: Module.symvers
