diff -u -r utadem.orig/Makefile utadem/Makefile
--- utadem.orig/Makefile	2008-09-25 22:26:46.000000000 +0200
+++ utadem/Makefile	2008-11-24 23:09:58.000000000 +0100
@@ -23,16 +23,16 @@
 
 COBJ = $(C_FILES:%.c=%.o)
 
-CFLAGS += -Wstrict-prototypes
-CFLAGS += -O
-CFLAGS += -fno-strict-aliasing
-CFLAGS += -mpreferred-stack-boundary=4
-#CFLAGS += -march=`uname -m`
-CFLAGS += -falign-functions=4
-CFLAGS += -D_SCCSID
-CFLAGS += -Wall
-CFLAGS += -Wno-unknown-pragmas
-CFLAGS += -DUTDEBUG
+EXTRA_CFLAGS += -Wstrict-prototypes
+EXTRA_CFLAGS += -O
+EXTRA_CFLAGS += -fno-strict-aliasing
+EXTRA_CFLAGS += -mpreferred-stack-boundary=4
+#EXTRA_CFLAGS += -march=`uname -m`
+EXTRA_CFLAGS += -falign-functions=4
+EXTRA_CFLAGS += -D_SCCSID
+EXTRA_CFLAGS += -Wall
+EXTRA_CFLAGS += -Wno-unknown-pragmas
+EXTRA_CFLAGS += -DUTDEBUG
 
 obj-m := $(COBJ)
 
diff -u -r utadem.orig/utadem.h utadem/utadem.h
--- utadem.orig/utadem.h	2008-09-25 22:26:46.000000000 +0200
+++ utadem/utadem.h	2008-11-24 23:15:55.000000000 +0100
@@ -14,7 +14,10 @@
 
 #if defined(CONFIG_COMPAT) || defined(CONFIG_SPARC64) || defined(CONFIG_X86_64) || defined(CONFIG_PPC64)
 #include <asm/uaccess.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #include <linux/ioctl32.h>
+#endif
 #include <linux/compat.h>
 #endif	/* CONFIG_COMPAT */
 
